implementations/rust/ockam/ockam_command/src/credential/issue.rs

```diff
@@ -1,24 +1,25 @@
 use ockam_core::compat::collections::HashMap;
 use ockam_core::compat::sync::Arc;
+use ockam_vault::Vault;
 
 use crate::{
     identity::default_identity_name,
     util::{node_rpc, print_encodable},
     vault::default_vault_name,
     CommandGlobalOpts, EncodeFormat, Result,
 };
-use anyhow::Context as _;
+use anyhow::{anyhow, Context as _};
 use clap::Args;
 use ockam::Context;
-use ockam_identity::{credential::CredentialBuilder, IdentityIdentifier};
+use ockam_identity::{credential::CredentialBuilder, PublicIdentity};
 
 #[derive(Clone, Debug, Args)]
 pub struct IssueCommand {
     #[arg(long = "as", default_value_t = default_identity_name())]
     pub as_identity: String,
 
     #[arg(long = "for", value_name = "IDENTITY_ID")]
-    pub for_identity: IdentityIdentifier,
+    pub for_identity: String,
```

Comment:
I would suggest to put instead `for_identity: PublicIdentity` here and specify a `value_parser=parse_public_identity` in the `clap` macro attribute. This way the parser can be reused everywhere we pass a `PublicIdentity` on the command line and we can avoid the lines 53 to 59 and similar lines in `store.rs`, `verify.rs`.

This would also make the possibility to provide a clear error message like `expecting a hex-encoded public identity. It can be produced with `ockam identity show --full --output hex`. Since we are modifying the API of the command line that can help people who were used to pass a simple identity identifier.

Reply:
I would love to do this; and that's how I originally started it.

However, I believe the only way to parse from a string to a Public Identity requires async calls and I was unable to see a way to use clap parsers with async functions.

Reply:
You're absolutely right, that is not possible out of the box at the moment. _However_ the parsing of a `hex` identity as a `PublicIdentity` should be possible without needing any async. We only need to use `async` at the moment because of how we decided to design our interfaces to the `Vault`. This is something I would like to revisit after the "identity PR" is merged.

Reply:
I think in the meantime it could still be a good idea to refactor a bit to avoid duplicating `pub async fn public_identity(&self) -> Result<PublicIdentity>` in several `credential/` files and also make sure we provide a good error message in case the `hex` decoding or the identity import fails (ideally with a unit test, I know I'm piling work onto you :-))